lib/MultiCompiler.js

```diff
@@ -185,10 +186,24 @@ module.exports = class MultiCompiler extends Tapable {
 	}
 
 	watch(watchOptions, handler) {
+		if (this.running)
+			return handler(
+				new Error(
+					"You ran Webpack twice. Each instance only supports a single concurrent compilation at a time."
+				)
+			);
+
+		const finalHandler = (err, stats) => {
```

Comment:
This is not correct. The handler is called multiple times while watching and when called watching has not ended. Omit all changes to this `finalHandler`.